pkg/scheduler/objects/queue.go

```diff
@@ -296,39 +297,64 @@ func (sq *Queue) setTemplate(conf configs.ChildTemplate) error {
 	return nil
 }
 
-// UpdateSortType updates the sortType for the queue based on the current properties.
-func (sq *Queue) UpdateSortType() {
+// UpdateQueueProperties updates the queue properties defined as text
+func (sq *Queue) UpdateQueueProperties() {
 	sq.Lock()
 	defer sq.Unlock()
 	// set the defaults, override with what is in the configured properties
 	if sq.isLeaf {
 		// walk over all properties and process
 		var err error
-		sq.sortType = policies.Undefined
 		for key, value := range sq.properties {
-			if key == configs.ApplicationSortPolicy {
+			switch key {
+			case configs.ApplicationSortPolicy:
 				sq.sortType, err = policies.SortPolicyFromString(value)
 				if err != nil {
 					log.Logger().Debug("application sort property configuration error",
 						zap.Error(err))
 				}
-			} else {
+				// if it is not defined default to fifo
+				if sq.sortType == policies.Undefined {
+					sq.sortType = policies.FifoSortPolicy
+				}
+			case configs.PreemptionPolicy:
+				sq.preemptionFenced, err = isFencingEnabled(value)
+				if err != nil {
+					log.Logger().Debug("queue fencing property configuration error",
+						zap.Error(err))
+				}
+			case configs.PreemptionDelay:
+				sq.preemptionDelay, err = time.ParseDuration(value)
+				if err != nil {
+					log.Logger().Debug("preemption delay property configuration error",
+						zap.Error(err))
+				}
```

Comment:
We need to ensure that preemptionDelay is positive, and default to 30s if not specified.

Reply:
Also, preemption policy needs to be supported on parent queues as well, so we can't just apply these in leaf queues.